src/Adapter/Doctrine/ORM/SearchCriteriaProvider.php

```diff
@@ -40,9 +40,14 @@ private function processSearchColumns(QueryBuilder $queryBuilder, DataTableState
             $column = $searchInfo['column'];
             $search = $searchInfo['search'];
 
-            if ('' !== trim($search) && null !== ($filter = $column->getFilter())) {
+            if ('' !== trim($search)) {
+                if (null !== ($filter = $column->getFilter())) {
+                    if (!$filter->isValidValue($search)) {
+                        continue;
+                    }
+                }
                 $search = $queryBuilder->expr()->literal($search);
```

Comment:
Need to build the search variable using the following:

$search = $queryBuilder->expr()->literal($column->getRightExpr($search));

Otherwise the % characters aren't added to the search string and partial matches won't happen.

Reply:
Also sometimes is necessary to use much more complicated comparisons in order to implement filtering with date range and such. CriteriaFilter with callback option would solve case like this.

Reply:
Wrapping `$column->getRightExpr($search)` in `literal` will stop `IN` operations from working